pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,183 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kubeclient "k8s.io/client-go/kubernetes"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+)
+
+const (
+	defaultGitTimeout = 20 * time.Second
+)
+
+type options struct {
+	git               git.Options
+	quickstartNameArg string
+	outputPath        string
+}
+
+func applyGitops(cmd *cmdutils.Cmd) {
+	cfg := api.NewClusterConfig()
+	cmd.ClusterConfig = cfg
+
+	cmd.SetDescription("apply", "Setting up GitOps and apply a Quick Start profile", "")
+
+	var opts options
+
+	cmd.SetRunFuncWithNameArg(func() error {
+		return doApplyGitops(cmd, opts)
+	})
+
+	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		fs.StringVarP(&opts.git.URL, "git-url", "", "", "URL for the git repository that will contain the cluster components")
+		fs.StringVarP(&opts.git.Branch, "git-branch", "", "master", "Git branch")
+		fs.StringVarP(&opts.outputPath, "output-path", "", "./", "Path to directory where the GitOps repo will be cloned")
+		fs.StringVar(&opts.git.User, "git-user", "Flux", "Username to use as Git committer")
+		fs.StringVar(&opts.git.Email, "git-email", "", "Email to use as Git committer")
+		fs.StringVar(&cfg.Metadata.Name, "cluster", "", "name of the EKS cluster to add the nodegroup to")
```

Comment:
Is this what `--name` is when using `create cluster`?

Reply:
Yes, but in this case the name could be the name of the quickstart profile so I went for `--cluster` like we do in `eksctl create nodegroup`. What do you think? I don't have a strong opinion here.

Reply:
Mh, ok. I wasn't aware of that.

I think it'd be nice if we had consistent options and flags across everything, but I guess cluster/name is not too confusing.

Reply:
I think @errordeveloper is the one who knows what it should be to be consistent.

Reply:
Yes, I think the current situation could be improved, but it's logical.

`--name` is used to refer to name of the things you are looking at, most of the time it's the name of the cluster, but in the case of nodegroups it's the name of the nodegroup.

It's also always the case that name is equivalent to argument:
- `eksctl create cluster foo` is equivalent to `eksctl create cluster --name=foo` 
- `eksctl create ng --cluster=foo bar` is equivalent to `eksctl create ng --cluster=foo --name=bar`

I'd like to depart from this, but renaming existing flags is hard and I'd rather see this exercise being done  on something simpler at first.

If you are going to have multiple profiles per cluster, and/or you want to allow user to specify profile name as argument - use `--name` for it.

Also, this command is still experiment, so we can change it later.

Reply:
Ok, then, for now I will keep the profile in the `--qiuckstart-profile` argument like Kalbir proposed in the issue, and the cluster name under `--cluster`. Then when he's back we can decide if we use the name argument and `--name` flag